climada/hazard/tc_tracks.py

```diff
@@ -1480,6 +1478,63 @@ def _xr_to_netcdf_multi(path, ds_dict, encoding=None):
     finally:
         store.close()
 
+def _xr_open_dataset_multi(path, prefix=""):
+    """Read multiple xarray Datasets from groups contained in a single NetCDF4 file
+
+    The data is loaded into memory
+
+    Contrary to xarray's `open_dataset` functionality, this only supports the "netcdf4" engine
+    since the groups feature has been introduced by NetCDF version 4.
+
+    Parameters
+    ----------
+    path : str or Path
+        Path of the NetCDF file to read.
+    prefix : str, optional
+        If given, only read groups whose name starts with this prefix. Default: ""
+
+    Returns
+    -------
+    ds_dict : dict whose keys are group names and values are xr.Dataset
+        Each xr.Dataset in the dict is taken from the group identified by its key in the dict.
+        Note that an empty string ("") is a valid group name and refers to the root group.
+    """
+    path = str(pathlib.Path(path).expanduser().absolute())
+    store = NetCDF4DataStore.open(path, "r", "NETCDF4", None)
+    ds_dict = {}
+    try:
```

Comment:
Again, I think exceptions should be handled in some way.

Reply:
No, see my comment above.